src/tools/getDatasourceMetadata/datasourceMetadataUtils.ts

```diff
@@ -0,0 +1,81 @@
+import { GraphQLResponse } from '../../sdks/tableau/apis/metadataApi.js';
+import { MetadataResponse } from '../../sdks/tableau/apis/vizqlDataServiceApi.js';
+
+export function buildDataDictionary(
+  readMetadataResult: MetadataResponse,
+  listFieldsResult: GraphQLResponse,
+): any {
+  // create a dataDictionary that combines field data from
+  // readMetadata (VizQL Data Service API) and listFields (GraphQL Metadata API) results
+  // to optimize for LLM accuracy and reduce tokens in response.
+  const dataDictionary = {
+    fields: Array<any>(),
+  };
+
+  if (!readMetadataResult.data) {
+    // TODO: what should we do here?
+    return dataDictionary;
```

Comment:
Can we safely map `listFieldsResult` to the shape of the data dictionary even if the `readMetadataResult` is empty? If not, while it would great to always return an object of the same shape from a tool, it's not necessary.